### PR TITLE
Added bulk target operations: remove/delete

### PR DESCRIPTION
**Pull request by mikefitz888**
Added functionality to allow the removal/deletion of all marked targets. This also included making the `toggleAll()` function only affect visible rows (so filtering works as expected), and check-boxes remain checked after changing pages on the table.

## Description
Added button to remove all selected targets from current session.
Added button to delete all selected targets.
Successes and failures are tracked, individual reporting is possible but not implemented.
TargetTable now keeps track of checked inputs between pagination.
Using the 'select all' checkbox will only toggle visible inputs.
Made the delete/remove target functions more flexible, functionality unchanged.

## Related Issue
https://github.com/owtf/owtf/issues/429

## Motivation and Context
This change makes it faster to remove/delete multiple targets. They don't need to be operated on individually. For example, to remove all low severity targets, you could filter by severity, select all, and use the add 'remove all' button.

## Reviewers
@delta24 

## How Has This Been Tested?
Changes have been tested and work on IceWeasel/FireFox and Chrome.

## Screenshots (if appropriate):
Before:
screenshot from 2016-04-02 13-40-46
After:
screenshot from 2016-04-02 13-36-50

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other

### Checklist:

- [ ] My code follows the code style (modified PEP8) of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by delta24** [comment]
@mikefitz888 Make the "0 targets selected" row invisible and toggle to show it only if at least one target is selected. For reference, see the zest integration (selecting HTTP transactions from the table).

**Comment by mikefitz888** [comment]
@delta24 fixed :)

File: framework/interface/templates/target_manager.html [hunk 1]
```diff
@@ -121,23 +130,37 @@
                     targets_search_api_url:"{{ targets_search_api_url }}",
                     targets_ui_url:"{{ targets_ui_url }}",
                     plugins_api_url:"{{ plugins_api_url }}?group=web&group=network",
-                    worklist_api_url:"{{ worklist_api_url }}"
+                    worklist_api_url:"{{ worklist_api_url }}",
+                    checked_rows:[] //Track checked rows for bulk operations
```
Comment (delta24) [inline framework/interface/templates/target_manager.html]: This is used to keep the constants. Move the variable `checked_rows` to somewhere else.

File: framework/interface/templates/target_manager.html [hunk 2]
```diff
@@ -121,23 +130,37 @@
...
-// Toggle all checkboxes when select all is clicked
+function updateSelectionCount(){
+    $('#targetTable .selected-target-count').closest('tr').css('display', mySpace.checked_rows.length ? 'table-row' : 'none');
+    $('#targetTable .selected-target-count').html(mySpace.checked_rows.length);
+}
+
+//Ensure an element is (state=true) or is not (state=false) in array 
+function setArrayElement(array, element, state) {
+    var index = array.indexOf(element);
+    if (index === -1 && state)
```
Comment (delta24) [inline framework/interface/templates/target_manager.html]: Curly braces in the `if` statement.

File: framework/interface/templates/target_manager.html [hunk 3]
```diff
@@ -121,23 +130,37 @@
...
+                    checked_rows:[] //Track checked rows for bulk operations
                   };
 
-// Toggle all checkboxes when select all is clicked
+function updateSelectionCount(){
+    $('#targetTable .selected-target-count').closest('tr').css('display', mySpace.checked_rows.length ? 'table-row' : 'none');
+    $('#targetTable .selected-target-count').html(mySpace.checked_rows.length);
+}
+
+//Ensure an element is (state=true) or is not (state=false) in array
```
Comment (delta24) [inline framework/interface/templates/target_manager.html]: Space after "//" and the actual comment.

**Comment by delta24** [comment]
@mikefitz888 a small bug: Say you have only 1 target. You select it and remove it/delete it. The line "1 targets currently selected." still remains. Expected behaviour is that no such message should be displayed.

Also change the success and error message to only have "x targets deleted/removed". Do not show "0 failed" if none failed. :)

Otherwise the functionality is correctly implemented albeit some styling issues are still there.

**Comment by mikefitz888** [comment]
@delta24 Thanks for reviewing that. I tidied any styling issues I could find, fixed the bug, and made the recommended changes.

**Pull request merged**